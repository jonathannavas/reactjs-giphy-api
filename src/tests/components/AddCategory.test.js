const { shallow } = require("enzyme");
const { AddCategory } = require("../../components/AddCategory");

describe('Test on AddCategory.js', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories} /> );

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories} /> );
    });

    test('Test to match snapshot', () => {
        expect( wrapper ).toMatchSnapshot()
    });

    test('Test on handleInputChange', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe( value );

    });

    test('Dont post info with de submit form ', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('should call setCategories and clear the input', () => {

        const value = 'Hola mundo';
        wrapper.find('input').simulate('change', { target:{ value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').prop('value') ).toBe('');

    });
    
});