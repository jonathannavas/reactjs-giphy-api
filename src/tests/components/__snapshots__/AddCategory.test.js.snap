// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on AddCategory.js Test to match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Ingrese un valor"
    type="text"
    value=""
  />
</form>
`;
